### PR TITLE
Rsz add h vrc set wire rc

### PR DESCRIPTION
**Pull request by arthurjolo**
Fixes [#3236](https://github.com/The-OpenROAD-Project/OpenROAD/issues/3236) 

This PR updates the command `set_wire_rc` to separate the values used to estimate wire parasitics for vertical and horizontal wires. The command now can receive a list of layers using the key `-layers`, the resistance and capacitance values of the layers passed will be used for wires according to their preferred direction. If more than one layer passed has the same preferred direction the average value will be used. 

Also the keys `-h_resistance -h_capacitance -v_resistance -v_capacitance` can be used to set these values directly.

### CODE REVIEW

File: src/rsz/test/rsz_readme_msgs_check.ok [hunk 1]
```diff
@@ -1,3 +1,3 @@
 README.md
 Names: 17,        Desc: 17,        Syn: 17,        Options: 17,        Args: 17
-Info: 35, Warn: 21, Error: 26
+Info: 35, Warn: 23, Error: 28
```
Comment (maliberty) [inline src/rsz/test/rsz_readme_msgs_check.ok:3]: @luarss this should not be necessary

File: src/rsz/src/Resizer.tcl [hunk 2]
```diff
@@ -125,49 +125,138 @@ proc set_layer_rc {args} {
...
+    set total_h_wire_cap 0.0
+    set total_v_wire_res 0.0
+    set total_v_wire_cap 0.0
+
+    set h_layers 0
+    set v_layers 0
+
+    set layers $keys(-layers)
+
+    foreach layer_name $layers {
```
Comment (maliberty) [inline src/rsz/src/Resizer.tcl:171]: There should be a check that exactly two layers were specified - one H and one V

Comment (arthurjolo) [reply src/rsz/src/Resizer.tcl:171]: I was following the comments on the issue that if more then one layer for a direction is passed we get the average values, should I change for only receiving 2 layers?

File: src/rsz/README.md [hunk 3]
```diff
@@ -44,9 +61,14 @@ set_wire_rc
...
 | `-resistance` | Resistance per unit length, units are from the first Liberty file read, usually in the form of $\frac{resistanceUnit}{distanceUnit}$. Usually kΩ/µm. |
 | `-capacitance` | Capacitance per unit length, units are from the first Liberty file read, usually in the form of $\frac{capacitanceUnit}{distanceUnit}$. Usually pF/µm. |
+| `-h_resistance` | Resistance per unit length for horizontal wires, units are from the first Liberty file read, usually in the form of $\frac{resistanceUnit}{distanceUnit}$. Usually kΩ/µm. |
+| `-h_capacitance` | Capacitance per unit length for horizontal wires, units are from the first Liberty file read, usually in the form of $\frac{capacitanceUnit}{distanceUnit}$. Usually pF/µm. |
+| `-v_resistance` | Resistance per unit length for vertical wires, units are from the first Liberty file read, usually in the form of $\frac{resistanceUnit}{distanceUnit}$. Usually kΩ/µm. |
+| `-v_capacitance` | Capacitance per unit length for vertical wires, units are from the first Liberty file read, usually in the form of $\frac{capacitanceUnit}{distanceUnit}$. Usually pF/µm. |
```
[the remark this answers is not in the document]
Comment (arthurjolo) [reply src/rsz/README.md:71]: Ok, will take that down

File: src/rsz/src/Resizer.tcl (hunk 2 above)
Comment (maliberty) [reply src/rsz/src/Resizer.tcl:171]: That's ok but we should ensure there is at least one H and one V specified.

Comment (arthurjolo) [reply src/rsz/src/Resizer.tcl:171]: Ok!

**Comment by maliberty** [comment]
Is this ready to go?  There is a conflict to resolve.

**Review by maliberty — APPROVED** [review]

**Pull request merged**